x/bridge/observer/observer.go

```diff
@@ -116,31 +139,69 @@ func (o *Observer) sendOutbound(ctx context.Context) {
 
 	for srcId, queue := range o.outTxQueue {
 		srcChain := o.chains[srcId]
-		height, err := srcChain.Height()
+		height, err := srcChain.Height(ctx)
 		if err != nil {
 			o.logger.Error(fmt.Sprintf("Failed to get height for %s: %s", srcId, err.Error()))
 			continue
 		}
-		confirmationsRequired, err := srcChain.ConfirmationsRequired()
-		if err != nil {
-			o.logger.Error(fmt.Sprintf(
-				"Failed to get confirmations required for %s: %s",
-				srcId,
-				err.Error(),
-			))
-			continue
-		}
-		var newQueue []Transfer
+
+		var newQueue []TxQueueItem
 		for _, out := range queue {
-			if height-out.Height < confirmationsRequired {
+			confirmed, err := o.isTxConfirmed(ctx, height, &out)
+			if !confirmed || err != nil {
+				if err != nil {
+					o.logger.Error(fmt.Sprintf("Failed to confirm tx %s", err.Error()))
+				}
 				newQueue = append(newQueue, out)
 				continue
 			}
-			err = o.chains[out.DstChain].SignalInboundTransfer(ctx, out)
+			err = o.chains[out.Tx.DstChain].SignalInboundTransfer(ctx, out.Tx)
 			if err != nil {
 				newQueue = append(newQueue, out)
 			}
 		}
 		o.outTxQueue[srcId] = newQueue
 	}
 }
+
+func (o *Observer) getConfirmationsRequired(ctx context.Context, tx Transfer) (uint64, error) {
+	if tx.SrcChain == ChainIdOsmosis {
+		return 0, nil
+	}
+
+	chain, ok := o.chains[ChainIdOsmosis]
+	if !ok {
+		return 0, fmt.Errorf("Chain client for %s not found", ChainIdOsmosis)
+	}
+	cr, err := chain.ConfirmationsRequired(ctx, bridgetypes.AssetID{
+		SourceChain: string(tx.SrcChain),
+		Denom:       tx.Asset,
+	})
+	if err != nil {
+		return 0, err
+	}
+	return cr, nil
+}
+
+func (o *Observer) isTxConfirmed(
+	ctx context.Context,
+	curHeight uint64,
+	item *TxQueueItem,
+) (bool, error) {
+	if curHeight < item.ConfirmationsRequired+item.Tx.Height {
+		return false, nil
+	}
+	cr, err := o.getConfirmationsRequired(ctx, item.Tx)
+	if err != nil {
+		return false, errorsmod.Wrapf(
+			err,
+			"Failed to get confirmations required for outbound transfer %s",
+			item.Tx.Id,
+		)
+	}
+	if curHeight < cr+item.Tx.Height {
+		item.ConfirmationsRequired = cr
+		return false, nil
+	}
```

Comment:
I don't understand what is happening here. It feels like 

curHeight < item.ConfirmationsRequired+item.Tx.Height

is the same check as

if curHeight < cr+item.Tx.Height

If its not we need explanations with in line comments and/or godocs